tools/roslaunch/env-hooks/10.roslaunch.bat

```diff
@@ -1,5 +1,12 @@
 REM roslaunch/env-hooks/10.roslaunch.bat
 
+:: workaround Python 2 xmlrpc performance issue
+:: https://stackoverflow.com/questions/2617615/slow-python-http-server-on-localhost
+:: use IP address instead to avoid unnecessary DNS lookups.
 if "%ROS_MASTER_URI%" == "" (
-  set ROS_MASTER_URI=http://localhost:11311
+  set ROS_MASTER_URI=http://127.0.0.1:11311
+)
+
+if "%ROS_IP%" == "" (
```

Comment:
This doesn't take into account if `ROS_HOSTNAME` is set. And it is discourage to set both - `ROS_IP` and `ROS_HOSTNAME` at the same time.

Reply:
@dirk-thomas Thanks. I edited the check to make sure `ROS_IP` is only set when both don't exist in the environments.